base/cvd/cuttlefish/host/commands/cvdalloc/privilege.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "cuttlefish/host/commands/cvdalloc/privilege.h"
+
+#include <errno.h>
+
+#if defined(__linux__)
+#include <linux/capability.h>
+#include <linux/prctl.h>
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+#endif
+
+#include <android-base/logging.h>
+
+#if defined(__linux__)
```

Comment:
Optional: An alternative to the larger `defined(__linux__)` blocks is to separate the files on operating system boundaries and have bazel choose which ones to build through a [`select` statement](https://bazel.build/extending/platforms). Something like

```
cc_library(
  name = "cvdalloc",
  srcs = ["cvdalloc.cpp", "privilege.cpp"] + select({
    "@platforms//os:linux": ["privilege_linux_capabilities.cpp"],
    "//conditions:default": ["privilege_setuid_default.cpp"],
  }),
  ...
)
```

This works better when the blocks cover entire functions rather than fragments of functions. In this case, the files could provide functions with the same signatures so the callers could be the same on different platforms.